tests/mhd/test_spec.py

```diff
@@ -292,6 +434,11 @@ def test_residue(self):
 
         # Initialize SPEC from an input file
         spec = Spec(filename)
+
+        # set axis to zero
+        spec.axis['rac'][:] = 0
+        spec.axis['zas'][:] = 0
+
```

Comment:
In the old version of `spec.py`, the coordinate axis given as input to SPEC, i.e. `rac` and `zas`, were ignored. Instead, these arrays were filled with zeros and SPEC constructed its own guess for the coordinate axis.

I modified `spec.py` to read the initial guess in SPEC input file. See [these lines](https://github.com/abaillod/simsopt/blob/4b4ba1ba98ae631f67dccbc9e6b958d19d62aa4e/src/simsopt/mhd/spec.py#L210-L220) and [these lines](https://github.com/abaillod/simsopt/blob/4b4ba1ba98ae631f67dccbc9e6b958d19d62aa4e/src/simsopt/mhd/spec.py#L792-L798).

When using the initial guess for the axis in `QH-residues.sp`, the `residue1` differs by about 0.5% from the value in the test. Another fix would be to modify `tests/test_files/QH-residue.sp` and remove the axis from it. A similar fix had to be made to `examples/2_Intermediate/inputs/QH-residues.sp` so that the example `examples/2_Intermediate/eliminate_magnetic_islands.py` still works. Let me know what you think is best!

Reply:
> When using the initial guess for the axis in `QH-residues.sp`, the `residue1` differs by about 0.5% from the value in the test. Another fix would be to modify `tests/test_files/QH-residue.sp` and remove the axis from it. A similar fix had to be made to `examples/2_Intermediate/inputs/QH-residues.sp` so that the example `examples/2_Intermediate/eliminate_magnetic_islands.py` still works. Let me know what you think is best!

Got it. I'm indifferent then between 3 options: (1) including those lines to set the axis to 0, (2) editing `QH-residue.sp` so the axis is 0 there, or (3) loosening the tolerance of the test so it passes without setting the axis to 0. For option (1), it might be useful to state in a comment why those lines are there.